spec/controllers/email_authentications/passwords_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+RSpec.describe EmailAuthentications::PasswordsController, type: :controller do
+  before :each do
+    @request.env["devise.mapping"] = Devise.mappings[:email_authentication]
+  end
+
+  describe "#edit" do
+    it "renders the edit page when the reset token is valid" do
+      auth = build(:email_authentication)
+      token = auth.send_reset_password_instructions
+      put :edit, params: {reset_password_token: token}
+      expect(response).to render_template "edit"
+    end
+
+    it "responds successfully when the reset token is expired" do
+      auth = build(:email_authentication)
+      token = auth.send_reset_password_instructions
+      auth.update_attribute(:reset_password_sent_at, 7.hours.ago)
```

Comment:
Two suggestions here:

* can you use `update!` so that the save will raise if anything goes wrong?
* it isn't clear that `7.hours` would mark the token as expired, and that could become an issue if we change the expiration time later. Maybe something like:

```
expired_sent_time = Devise.config.reset_password_within + 1.hour
auth.update!(reset_password_sent_at: expired_sent_time)
```

Reply:
Yeah, I should've thought to use the Devise config. Good point.